src/CloudFoundry.CloudController.Test.Integration/LoggregatorTest.cs

```diff
@@ -0,0 +1,220 @@
+        }
+
+        static void Apps_PushProgress(object sender, PushProgressEventArgs e)
+        {
+            Console.WriteLine(e.Message + " " + e.Percent);
+        }
+
+        [TestMethod]
+        [TestCategory("RequiresLoggregator")]
+        public void LoggregatorRecentTest()
+        {
+            CreateAppResponse app = client.Apps.CreateApp(apprequest).Result;
+
+            Guid appGuid = app.EntityMetadata.Guid;
+
+            client.Apps.Push(appGuid, tempAppPath, true).Wait();
+
+            while (true)
+            {
+                var appSummary = client.Apps.GetAppSummary(appGuid).Result;
+                var packageState = appSummary.PackageState.ToLowerInvariant();
+
+                if (packageState != "pending")
+                {
+                    Assert.AreEqual(packageState, "staged");
+
+                    var instances = client.Apps.GetInstanceInformationForStartedApp(appGuid).Result;
+
+                    if (instances.Count > 0)
+                    {
+                        if (instances[0].State.ToLower() == "running")
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+
+            if (client.Info.GetInfo().Result.LoggingEndpoint == null)
+            {
+                Assert.Inconclusive("CloudFoundry target does not have a loggregator endpoint");
+            }
+
+            var logClient = new LoggregatorLog(
+                new Uri(client.Info.GetInfo().Result.LoggingEndpoint),
+                string.Format("bearer {0}", client.AuthorizationToken),
+                null,
+                true);
+
+            // Just wait a bit to get the latest logs
+            Thread.Sleep(1000);
+
+            var appLogs = logClient.Recent(appGuid.ToString(), CancellationToken.None).Result;
+
+            var conatainsPushedContent = appLogs.Any((line) => line.Message.Contains("dummy content"));
+            Assert.IsTrue(conatainsPushedContent, "Pushed content was not dumped in the output stream: {0}", string.Join(Environment.NewLine, appLogs.Select(x => x.Message)));
+
+            var conatainsEnvContent = appLogs.Any((line) => line.Message.Contains("envtest1234"));
+            Assert.IsTrue(conatainsEnvContent, "Pushed env variable was not dumped in the output stream: {0}", string.Join(Environment.NewLine, appLogs.Select(x => x.Message)));
+
+            client.Apps.DeleteApp(appGuid).Wait();
+            Directory.Delete(tempAppPath, true);
+        }
+
+        [TestMethod]
+        [TestCategory("RequiresLoggregator")]
+        public void LoggregatorTailTest()
+        {
+            CreateAppResponse app = client.Apps.CreateApp(apprequest).Result;
+
+            Guid appGuid = app.EntityMetadata.Guid;
+
+            if (client.Info.GetInfo().Result.LoggingEndpoint == null)
+            {
+                Assert.Inconclusive("CloudFoundry target does not have a loggregator endpoint");
```

Comment:
We've had issues with inconclusive - I remember using categories to decide which tests we want to run against a certain target?